### PR TITLE
Switch the CONTRIBUTING page to Asciidoc, add table of contents

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I am about documenting more practices n this page, including the code review process. So the page will be much bigger, and I would like to introduce table of contents for it. Hence, I propose shifting the page to Asciidoc

### CODE REVIEW

**Review by halkeye — APPROVED** [review]
Looks like just formatting no content changes time. So lgtm.

**Review by MarkEWaite — APPROVED** [review]

**Comment by res0nance** [comment]
Thinking out loud but perhaps for links such as `https://github.com/jenkinsci/jenkins/blob/c3373cb975d55193072caee65dc6f4cf6ab3398b/.github/PULL_REQUEST_TEMPLATE.md` it might be better to link directly from the master branch else the content might get stale. OTOH we would have to update the contributing file if these links were to move.

**Review by timja — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]

File: CONTRIBUTING.adoc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+The code for the core build flow is stored in the link:./Jenkinsfile[Jenkinsfile] in the repository root.
+If you want to update that build flow (e.g. "add more checks"),
+just submit a pull request.
+
+== Links
+
+* link:https://jenkins.io/participate/[Jenkins Contribution Landing Page]
+* link:https://jenkins.io/chat/[Jenkins IRC and Gitter Channels]
+* link:https://wiki.jenkins.io/display/JENKINS/Beginners+Guide+to+Contributing[Beginners Guide To Contributing]
+* link:(https://issues.jenkins-ci.org/issues/?jql=project%20%3D%20JENKINS%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20AND%20component%20%3D%20core%20AND%20labels%20in%20(newbie-friendly)[List of newbie-friendly issues in the core]
```
Comment (varyvol) [inline CONTRIBUTING.adoc]:
```suggestion
* link:https://issues.jenkins-ci.org/issues/?jql=project%20%3D%20JENKINS%20AND%20status%20in%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20AND%20component%20%3D%20core%20AND%20labels%20in%20(newbie-friendly)[List of newbie-friendly issues in the core]
```

**Review by varyvol — COMMENTED** [review]
The "Table of contents" heading is not highlighted, but it's not in this file. How is that generated?

**Comment by oleg-nenashev** [comment]
On hold till the end of Hacktoberfest + it is already conflicted. Will just close it and rework later, it is easy to migrate with Pandoc

**Pull request closed without merging**